docs/plugin_developers_guide/09_packaging.rst

```diff
@@ -0,0 +1,256 @@
+
+
+- service_notification.version
+   This file only requires if plugin is notification rule or delivery based.
+
+C based Plugins
+---------------
+
+- VERSION
+   Plugin version number.
+- fledge.version
+   Fledge version number.
+- requirements.sh (OPTIONAL)
+   If a plugin requires some additional libraries to install with. Also make sure file should be in executable format.
+
+Examples of filename along with content
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. VERSION
+
+.. code-block:: console
+
+    $ cat VERSION
+    1.9.2
+
+2. fledge.version
+
+.. code-block:: console
+
+    $ cat fledge.version
+    fledge_version>=1.9
+
+3. requirements.sh
+
+.. code-block:: console
+
+    $ cat requirements.sh
+    #!/usr/bin/env bash
+    which apt >/dev/null 2>&1
+    if [ $? -eq 0 ]; then
+        sudo apt install -y libmodbus-dev
+    else
+        which yum >/dev/null 2>&1
+        if [ $? -eq 0 ]; then
+            sudo yum -y install epel-release libmodbus libmodbus-devel
+        fi
+    fi
+
+4. Description
+
+.. code-block:: console
+
+    $ cat Description
+    Fledge modbus plugin. Supports modbus RTU and modbus TCP.
+
+5. Package
+
+.. code-block:: console
+
+    $ cat Package
+    # A set of variables that define how we package this repository
+    #
+    plugin_name=modbus
+    plugin_type=south
+    plugin_install_dirname=ModbusC
+    plugin_package_name=fledge-south-modbus
+    additional_libs="usr/local/lib:/usr/local/lib/libsmod.so*"
+
+    # Now build up the runtime requirements list. This has 3 components
+    #   1. Generic packages we depend on in all architectures and package managers
+    #   2. Architecture specific packages we depend on
+    #   3. Package manager specific packages we depend on
+    requirements="fledge"
+
+    case "$arch" in
+        x84_64)
+            ;;
+        armv7l)
+            ;;
+        aarch64)
+            ;;
+    esac
+    case "$package_manager" in
+        deb)
+            requirements="${requirements},libmodbus-dev"
+            ;;
+        rpm)
+            requirements="${requirements},epel-release,libmodbus,libmodbus-devel"
+            ;;
+    esac
+
+.. note::
+    If your package is not supported for a specific platform then you must exit with exitcode 1
+
+6. service_notification.version
+
+.. code-block:: console
+
+    $ cat service_notification.version
+    service_notification_version>=1.9.2
+
+Common Additional Libraries Package
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+At the moment we have only two package exists.
+
+- **fledge-mqtt** which is packed with libpaho-mqtt library.
+- **fledge-gcp** which is packed with libjwt and libjansson libraries.
+
+If your plugin depends upon then update **requirements** variable in Package file instead of using *additional_libs* variable.
+
+Python based Plugins
+--------------------
+
+- VERSION.{PLUGIN_TYPE}.{PLUGIN_NAME}
+   Fledge and Plugin version number.
```

Comment:
The *VERSION.{PLIGIN_TYPE}.{PLUGIN_NAME}* file contains the packaged version of the the plugin and also the fledge version that the plugin requires.  The format of this file is...